cpp/onedal/array.hpp

```diff
@@ -32,54 +34,155 @@ class array {
   friend array<Y> const_array_cast(const array<U>&);
 
   public:
```

Comment:
Have we decided to prohibit `array<const T>`? @michael-smirnov, what is your opinion? If decided, need to add static assert somewhere in class scope. I'd like to add it in the very beginning.

Reply:
Yes, I added `static_assert` clause.